policy/src/main/java/org/apache/brooklyn/policy/location/CreateLocationPolicy.java

```diff
@@ -0,0 +1,288 @@
+            .build();
+
+    @SuppressWarnings("serial")
+    public static final ConfigKey<Set<String>> LOCATION_TAGS = ConfigKeys.builder(new TypeToken<Set<String>>() {})
+            .name("location.tags")
+            .description("Tags for the created location")
+            .defaultValue(ImmutableSet.of())
+            .build();
+
+    private AtomicBoolean created = new AtomicBoolean(false);
+    private Object mutex = new Object[0];
+    private CatalogItem<?,?> catalogItem;
+    private AtomicInteger version = new AtomicInteger(0);
+
+    public CreateLocationPolicy() {
+        this(MutableMap.<String,Object>of());
+    }
+
+    public CreateLocationPolicy(Map<String,?> props) {
+        super(props);
+    }
+
+    protected AttributeSensor<Boolean> getStatusSensor() {
+        return config().get(LOCATION_STATUS);
+    }
+
+    protected Map<String,AttributeSensor<?>> getLocationConfiguration() {
+        return config().get(LOCATION_CONFIG);
+    }
+
+    protected String getLocationCatalogItemId() {
+        return config().get(LOCATION_CATALOG_ID);
+    }
+
+    protected String getLocationDisplayName() {
+        return config().get(LOCATION_DISPLAY_NAME);
+    }
+
+    protected String getLocationType() {
+        return config().get(LOCATION_TYPE);
+    }
+
+    protected Set<String> getLocationTags() {
+        return config().get(LOCATION_TAGS);
+    }
+
+    @Override
+    protected <T> void doReconfigureConfig(ConfigKey<T> key, T val) {
+        throw new UnsupportedOperationException("reconfiguring "+key+" unsupported for "+this);
+    }
+
+    private final SensorEventListener<Boolean> lifecycleEventHandler = new SensorEventListener<Boolean>() {
+        @Override
+        public void onEvent(SensorEvent<Boolean> event) {
+            synchronized (mutex) {
+                Boolean status = event.getValue();
+                if (status) {
+                    if (created.compareAndSet(false, true)) {
+                        LOG.info("Creating new location {} of type {}", getLocationCatalogItemId(), getLocationType());
+
+                        ImmutableList.Builder<String> builder = ImmutableList.<String>builder().add(
+                                "brooklyn.catalog:",
+                                "  id: " + getLocationCatalogItemId(),
+                                "  version: " + version.incrementAndGet(),
+                                "  itemType: location",
+                                "  item:",
+                                "    type: " + getLocationType(),
+                                "    brooklyn.config:",
+                                "      displayName: " + getLocationDisplayName());
+
+                        for (Map.Entry<String, AttributeSensor<?>> entry : getLocationConfiguration().entrySet()) {
+                            AttributeSensor<?> sensor = getLocationConfiguration().get(entry.getKey());
+                            Object value = entity.sensors().get(sensor);
```

Comment:
what happens if the sensor is not yet populated, wouldn't be safer to require directly value so we can use `attributeWhenReady` in the blueprint? or maybe we can always assume that once `LOCATION_STATUS` is true, brooklyn would have sensor data ready?